token/cli/src/main.rs

```diff
@@ -943,11 +953,34 @@ fn command_burn(
     })
 }
 
+
+fn check_if_amount_overflows_supply(  
+    config: &Config,
+    token: Pubkey,
+    ui_amount : u64,
+) -> bool {
+    let supply = config.rpc_client.get_token_supply(&token).unwrap();
+    let amount = ui_amount;
+    if amount > u64::MAX {
+        return true;
+    };
+    let supply_val = supply.amount;
+    let supply_u64: u64 = supply_val.parse().unwrap();
+    if amount.checked_add(supply_u64).is_none() {
+        return true;
+    };
+    if amount + supply_u64 > u64::MAX {
+    return true;
+    };
+    return false;
```

Comment:
nit: The check here is now being done twice -- you can condense this all down to just
```suggestion
    amount.checked_add(supply).is_none()
```